### PR TITLE
Readme improvements

### PR DESCRIPTION
**Pull request by stephenfin**
Subject: Improve the README

### Feature or Bugfix
Refactor

### Purpose
Move the contents of `long_desc` from `setup.py` into the README, and simply use this README for `long_desc`.

### Detail
- setup.cfg: Restructure file

  Place configuration of the package itself first in the file, followed by configuration of other tools.

- Improve README

  We're going to be adding additional badges to this but, before we do that, we should improve the README and remove a lot of the duplication that has built up here.

  The 'long_description' from 'setup.py' is moved into the README as there is serious value in displaying this on GitHub, as much as there is value in including the badges and other content from README.

- README: Cleanup badges

  Add AppVeyor badge and include 'alt' attributes for the others.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,45 +1,94 @@
...
-=================
-
-This is the Sphinx documentation generator, see http://www.sphinx-doc.org/.
-
-
-Installing
-==========
-
-Install from PyPI to use stable version::
+Sphinx is published on `PyPi`__ and can be installed from there::
```
Comment (tk0miya) [inline README.rst]: `PyPI` (last letter is uppercase)
https://pypi.python.org/pypi

File: README.rst [hunk 2]
```diff
@@ -1,45 +1,94 @@
...
+Documentation is available from `sphinx-doc.org`__.
 
-Install from cloned source::
+__ http://www.sphinx-doc.org/
 
-   pip install .
+Testing
+=======
+
+Continuous testing is provided by `Travis`__ and `AppVeyor`__. For information
```
Comment (tk0miya) [inline README.rst]: Note: we also uses Circle CI :-)
https://circleci.com/gh/sphinx-doc/sphinx

We use it for heavy process like TeX compilation and so on using Docker image.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits

File: README.rst (hunk 1 above)
Comment (stephenfin) [reply README.rst]: Good call. Done.

File: README.rst (hunk 2 above)
Comment (stephenfin) [reply README.rst]: Agreed. Done and expanded with reasons for why we do each thing.

File: README.rst (hunk 1 above)
Comment (tk0miya) [reply README.rst]: It seems not updated. Please push your commit.

Comment (tk0miya) [reply README.rst]: It seems not updated. Please push your commit.

Comment (stephenfin) [reply README.rst]: @tk0miya Oops. Now it's done.

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Thank you always!

**Comment by stephenfin** [comment]
@tk0miya It looks like the README [is not displaying links correctly](https://github.com/sphinx-doc/sphinx/blob/b00f78311fd359435c7082e7e3b7590c05fb005b/README.rst). Do you know why this is? I thought `rst2html` supported all these link styles. If not, I will work on this shortly and submit a fix once I identify the issue.

**Comment by stephenfin** [comment]
> @tk0miya It looks like the README is not displaying links correctly. Do you know why this is? I thought rst2html supported all these link styles. If not, I will work on this shortly and submit a fix once I identify the issue.

Fixed in #4326

**Comment by tk0miya** [comment]
Sorry, I don't know. Perhaps, github does not support the notation...

**Comment by tk0miya** [comment]
Oh, I've overlooked last your comment. It seems fine at current master.

**Pull request merged**